libs/payments/cart/src/lib/checkout.service.ts

```diff
@@ -384,27 +374,135 @@ export class CheckoutService {
+            cartId,
+          });
+        }
+        const subscription = await this.subscriptionManager.retrieve(
+          cart.stripeSubscriptionId
+        );
+        if (!subscription) {
+          throw new CartSubscriptionNotFoundError(cartId);
+        }
+
+        const customer = cart.stripeCustomerId
+          ? await this.customerManager.retrieve(cart.stripeCustomerId)
+          : undefined;
+        if (!customer) {
+          throw new CartError('Customer not found for cart', { cartId });
+        }
+
+        // Process PayPal payment
+        if (subscription.collection_method === 'send_invoice') {
+          if (!subscription.latest_invoice) {
+            throw new CartError('No latest invoice found for subscription', {
+              subscriptionId: subscription.id,
+            });
+          }
+          try {
+            const latestInvoice = await this.invoiceManager.retrieve(
+              subscription.latest_invoice
+            );
+            await this.invoiceManager.processPayPalInvoice(latestInvoice);
+            await this.postPaySteps(cart, cart.version, subscription, cart.uid);
+          } catch (e) {
+            console.error(e);
+            await this.subscriptionManager.cancel(subscription.id);
+            await this.paypalBillingAgreementManager.cancel(
+              customer.metadata[STRIPE_CUSTOMER_METADATA.PaypalAgreement]
+            );
+            await this.cartManager.finishErrorCart(cartId, {
+              errorReasonId: CartErrorReasonId.Unknown,
+            });
+            return;
+          }
+        } else {
+          // process Stripe payment
+          const paymentIntent =
+            await this.subscriptionManager.getLatestPaymentIntent(subscription);
+          if (!paymentIntent) {
+            throw new CartError('PaymentIntent not found for subscription', {
+              subscriptionId: subscription.id,
+            });
+          }
+          switch (paymentIntent.status) {
+            case 'requires_action':
+              await this.cartManager.setNeedsInputCart(cartId);
+              return;
+            case 'succeeded':
+              if (paymentIntent.payment_method) {
+                await this.customerManager.update(customer.id, {
+                  invoice_settings: {
+                    default_payment_method: paymentIntent.payment_method,
+                  },
+                });
+              } else {
+                throw new CartError(
+                  'Failed to update customer default payment method',
+                  { cartId }
+                );
+              }
+              await this.postPaySteps(
+                cart,
+                cart.version,
+                subscription,
+                cart.uid
+              );
+              return;
+            case 'requires_capture':
+            case 'requires_confirmation':
+            case 'requires_payment_method':
+            case 'processing':
```

Comment:
At this stage of checkout, all statuses except for `processing` should result in this method throwing an error. I provide some reasoning for each status below.

- `requires_capture` - SubPlat doesn't currently separately [authorizing and capturing funds](https://docs.stripe.com/payments/place-a-hold-on-a-payment-method)
- `requires_confirmation` - At this point in checkout, the payment intent should already have been confirmed.
- `requires_payment_method` - At this point in checkout a payment method should already be attached.

`processing` - Technically at this point we don't need polling logic on `processing` either, since [Card payments don't make use of this status](https://docs.stripe.com/payments/paymentintents/lifecycle#intent-statuses). However, if its necessary for some payment methods we'll be adding in the near future, then it's OK keeping it in here.
> Other payment methods, such as cards, are processed more quickly and don’t go into the processing status.